src/packages/core/property/property/property.element.ts

```diff
@@ -234,7 +234,10 @@ export class UmbPropertyElement extends UmbLitElement {
 				this.#valueObserver = this.observe(
 					this.#propertyContext.value,
 					(value) => {
+						// Set the value on the element:
 						this._element!.value = value;
+						// Set the value on the context as well, to ensure that any default values are stored right away:
+						this.#propertyContext.setValue(value);
```

Comment:
When reading this code I get suspicious that something else is wrong.

Like what I read happening here is that the observation callback is called when the propertyContext value changes, and in that callback the value gets set again? Like why would it need to set the value again, if it was just set.

I'm not sure how that could make any difference, but if so I would say something else should be implemented differently.

In terms of the preset data, like in your example, it does to me sound as if the data of the preset isn't begin set to the workspace data? and this code might trigger something so that happens, but this is not good from a data observation point.

Reply:
That might very well be. I think this might just have been an oversight. As you see on the line immediately above, the value is sent to the element `this._element!.value` so the element knows about initial values. However, the element will never inform the property context of those values before the user touches them. That is why I thought if we just gave the context the initial value as well, then all values would count as "touched" and thereby be saved.

It's another story entirely if we don't want to persist initial values on purpose, then we need to let the elements know that they should fallback to hardcoded initial values if the value is undefined.

What do you think, do you want to look more into it?

Reply:
We are sort of missing a default value system. But I think that is a feature out of our scope.

So to get around this I made a PR to change the way the default are set in the Data-Type Workspace.
Just cause I'm affright this code could lead to other problems in the future.